File: src/client/java/com/mod/client/compat/ScreenGraphics.java
package com.mod.client.compat;

import net.minecraft.client.gui.Font;
import net.minecraft.network.chat.Component;

public interface ScreenGraphics {
    void fill(int x1, int y1, int x2, int y2, int color);
    void text(Font font, Component text, int x, int y, int color, boolean shadow);
}

File: src/client/kotlin/com/mod/client/compat/ScreenGraphics.kt
package com.mod.client.compat

import net.minecraft.client.gui.Font
import net.minecraft.network.chat.Component

interface ScreenGraphics {
    fun fill(x1: Int, y1: Int, x2: Int, y2: Int, color: Int)
    fun text(font: Font, text: Component, x: Int, y: Int, color: Int, shadow: Boolean)
    fun enableScissor(x1: Int, y1: Int, x2: Int, y2: Int)
    fun disableScissor()
}
